aicsimageio/readers/reader.py

```diff
@@ -331,9 +331,39 @@ def get_image_data(
     @property
     @abstractmethod
     def dims(self) -> str:
+        # When implementing a new reader you must use
+        # @Reader.dims.getter instead of simply @property
+        # See:
+        # https://stackoverflow.com/questions/15785982/python-overriding-getter-without-setter/15786149#15786149
         pass
 
-    def size(self, dims: str = Dimensions.DefaultOrder) -> Tuple[int]:
+    @dims.setter
+    def dims(self, dims: str):
+        """
+        Parameters
+        ----------
+        dims: str
+            The dimensions of the image.
+
+        Raises
+        ------
+        exceptions.InvalidDimensionOrderingError:
+            If too many or too few dimensions were provided in comparison to the
+            available image data.
+        """
+        # Check amount of provided dims against data shape
+        if len(dims) != len(self.dask_data.shape):
+            raise exceptions.InvalidDimensionOrderingError(
+                f"Provided too many dimensions for the associated file. "
+                f"Received {len(dims)} dimensions [dims: {dims}] "
+                f"for image with {len(self.data.shape)} dimensions "
+                f"[shape: {self.data.shape}]."
+            )
+
+        # Set the dims
+        self._dims = dims
+
+    def get_size(self, dims: str = Dimensions.DefaultOrder) -> Tuple[int]:
```

Comment:
This function and `get_size_of_dimension` could be considered for renaming imo.

Reply:
Should this be `get_shape` instead?